src/Proto.Remote/RemoteProcess.cs

```diff
@@ -23,35 +23,33 @@ public RemoteProcess(ActorSystem system, EndpointManager endpointManager, PID pi
 
         private void Send(object msg)
         {
+            object message;
+            var endpoint = _endpointManager.GetEndpoint(_pid.Address);

```

Comment:
Wouldn't it be possible to keep this `endpoint` reference around after the first lookup?
Meaning it only performs the dict lookup once.

It might have to be cleared out if the endpoint have been terminated and later recreated, but might be doable?

Reply:
As you suggested, I'm now storing the endpoint pid in the remoteProcess. But it's done only if it's not null, keeping the current behavior when it tries to connect to a terminated endpoint.

To handle the case when the endpoint is terminated, I added a subscription to the DeadLetterEvent in the EndpointManager to handle the RemoteWatch and RemoteDeliver messages to translate them.